### PR TITLE
Codeclimate to monitor coverage and technical debt

### PR DESCRIPTION
**Pull request by soumeh01**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -150,9 +150,11 @@ jobs:
         run: |
           make coverage-check
 
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+      - name: Publish coverage report to Code Climate
```
Comment (jkrech) [inline .github/workflows/test.yml:153]: So strictly speaking we are replacing "codecov" with "codeclimate" providing feedback on technical debt analysis in addition to testing coverage.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -150,9 +150,11 @@ jobs:
...
+        uses: paambaati/codeclimate-action@v5.0.0
```
Comment (jkrech) [inline .github/workflows/test.yml:154]: I am always a bit nervous using an action from a user. How reliable is that and what is the risk of having this dependency. Is there something that checks the actions we are using?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -150,9 +150,11 @@ jobs:
...
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
         with:
-          files: ./build/cover.out
-          fail_ci_if_error: true
-          functionalities: fix
+          debug: true
```
Comment (jkrech) [inline .github/workflows/test.yml:158]: What is controlled by "debug" ? Is this providing debug information or is it consuming debug info we provide?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -150,9 +150,11 @@ jobs:
...
+          coverageLocations: ./cover.out:gocov
```
Comment (jkrech) [inline .github/workflows/test.yml:159]: I assume you have tested this, however it looks like the location of cover.out changed from ./build/cover.out to ./cover.out probably due to the different actions we are using?

**Review by jkrech — DISMISSED** [review]
I am lacking knowledge to assess the changes in detail but agree to the switch from codecov to codeclimate.

File: .github/workflows/test.yml (hunk 1 above)
Comment (soumeh01) [reply .github/workflows/test.yml:153]: For the time being Yes. In the future, we will try to configure go format and go linting using codeclimate itself.

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**